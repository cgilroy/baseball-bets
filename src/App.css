* {
  box-sizing: border-box;
}

.App {
  text-align: center;
}

.App-header {
  background-color: #282c34;
  height: 100px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.lds-ripple {
  display: inline-block;
  position: relative;
  width: 64px;
  height: 64px;
}
.lds-ripple div {
  position: absolute;
  border: 4px solid #282c34;
  opacity: 1;
  border-radius: 50%;
  animation: lds-ripple 1s cubic-bezier(0, 0.2, 0.8, 1) infinite;
}
.lds-ripple div:nth-child(2) {
  animation-delay: -0.5s;
}
@keyframes lds-ripple {
  0% {
    top: 28px;
    left: 28px;
    width: 0;
    height: 0;
    opacity: 1;
  }
  100% {
    top: -1px;
    left: -1px;
    width: 58px;
    height: 58px;
    opacity: 0;
  }
}

@media only screen and (max-width:600px) {
  .win-count {
    display: none;
  }
  .checkmarkIcon {
    display: none;
  }
}

.win-count {

}
