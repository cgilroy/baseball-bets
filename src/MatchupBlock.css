/* CSS for MatchupBlock.js */
.gamesSection__header {
  margin: 5px 0;
}
.card {
  width: 49%
}
.card__content {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
}
.bullet {
  display: inline-block;
  margin: 0 2px;
  transform: scale(0.8);
}
.title {
  font-size: 14px;
}
.pos {
  margin-bottom: 12px;
}
.matchup {
  display: flex;
  justify-content: space-between;
  align-items: center
}
.team-logo {
  height: 25px;
  padding-right: 8px;
}
.win-count {
  padding: 0 8px;
}
.card__winner {
  border: 2px solid #00d064;
  box-sizing: border-box
}
.card__content--game-loser {
  color: grey
}
.card__content--game-loser .team-logo {
    filter: grayscale(100%);
}
.dropDown {
  display: flex;
  justify-content: center;
  flex-wrap: wrap
}
.collapsed {
  transition: all .15s linear
}
.expanded {
  transition: all .15s linear;
  transform: rotate(180deg)
}
.centerScore {
  display: flex;
  align-items: center;
  width: 150px;
  justify-content: space-around;
}
.liveIcon {
  padding-right: 4px;
  animation: liveLoop 0.4s ease-in-out alternate infinite;
}
.liveIndicator {
  position: relative;
  height: 2px;
  overflow: hidden;
  margin-top: 3px;
}
.liveIndicator__bar {
  height: 100%;
  width: 40px;
  position: absolute;
  background: green;
  animation: loopBar 4s 1s infinite alternate ease-in-out
}

@keyframes loopBar {
  0% {
    left: -40px
  }
  60% {
    left: -40px
  }
  100% {
    left: 100%;
  }

}

@keyframes liveLoop {
  0% {
    opacity: 1
  }
  100% {
    opacity: 0
  }

}

@media only screen and (max-width:600px) {
  .matchup {
    flex-wrap: wrap;
  }
  .card {
    height: 75px;
    align-items: center;
    display: flex
  }
  .centerScore {
    width: 100%;
    order: 3;
    margin-top: 10px;
  }
  .checkmarkIcon {
    display: none;
  }
}
