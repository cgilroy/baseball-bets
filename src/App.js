import React from 'react';
import github from './resources/github.svg'
import './App.css';
import FetchData from './FetchData.js'
import MatchupBlock from './MatchupBlock.js'
import BettingSummary from './BettingSummary.js'
import useInterval from './useInterval.js'
import {useState, useEffect} from 'react'
import leagueLogo from './resources/league-logo.svg'

function App() {
  const [allData, setAllData] = useState()
  const [loading, setLoading] = useState(true)
  const [betObjects, setBetObjects] = useState([])
  var moment = require('moment')
  const [scheduleDate, setScheduleDate] = useState(new Date())

  const doneFetch = (data) => {
    // console.log('donefetch')
    if (loading) setLoading(false);
    setAllData(data)
  }

  const handleDateChange = (date) => {
    // console.log('schedChange',date)
    setBetObjects([])
    setScheduleDate(date)
    FetchData(moment(date).format('L'),doneFetch)
  }

  useEffect(() => {
    FetchData(moment(scheduleDate).format('L'),doneFetch)
  },[])

  useInterval(() => {
    FetchData(moment(scheduleDate).format('L'),doneFetch);
  }, 15000);

  var liveGames = []
  var scheduledGames = []
  var finalGames = []
  let betObjectsTemp = []
  var betSummary = <BettingSummary handleDateChange={handleDateChange} betData={betObjects} />

  if (allData !== undefined) {
    let schedule = allData.find(obj => obj.dataType === "schedule")
    let pitchingData = allData.find(obj => obj.dataType === "pitching").data
    let hittingData = allData.find(obj => obj.dataType === "hitting").data
    let fieldingData = allData.find(obj => obj.dataType === "fielding").data
    let startingPitcher = allData.find(obj => obj.dataType === "startingPitcherStats").data
    let recordData = allData.find(obj => obj.dataType === "records").data
    const addBetObject = (obj) => {
      betObjectsTemp.push(obj)
      setBetObjects(betObjectsTemp)
    }
    schedule.data.games.map(game => {
      let homePitcher = ''
      let awayPitcher = ''
      if (game.teams.home.hasOwnProperty('probablePitcher')) {
        homePitcher = (startingPitcher.find(obj => obj.id == game.teams.home.probablePitcher.id) !== undefined) ? (
          startingPitcher.find(obj => obj.id == game.teams.home.probablePitcher.id).stats
        ) : ('');
      }

      if (game.teams.away.hasOwnProperty('probablePitcher')) {
        awayPitcher = (startingPitcher.find(obj => obj.id == game.teams.away.probablePitcher.id) !== undefined) ? (
          startingPitcher.find(obj => obj.id == game.teams.away.probablePitcher.id).stats
        ) : ('');
      }

      let homeData = {
        id: game.teams.home.team.id,
        name: game.teams.home.team.name,
        records: recordData.find(obj => obj.id === game.teams.home.team.id),
        pitching: pitchingData.find(obj => obj.team.id === game.teams.home.team.id).stat,
        hitting: hittingData.find(obj => obj.team.id === game.teams.home.team.id).stat,
        fielding: fieldingData.find(obj => obj.team.id === game.teams.home.team.id).stat,
        pitcherStats: homePitcher
      }
      let awayData = {
        id: game.teams.away.team.id,
        name: game.teams.away.team.name,
        records: recordData.find(obj => obj.id === game.teams.away.team.id),
        pitching: pitchingData.find(obj => obj.team.id === game.teams.away.team.id).stat,
        hitting: hittingData.find(obj => obj.team.id === game.teams.away.team.id).stat,
        fielding: fieldingData.find(obj => obj.team.id === game.teams.away.team.id).stat,
        pitcherStats: awayPitcher
      }
      let gameState = game.status.codedGameState
      let outputBlock = <MatchupBlock key={game.gamePk} addBetObject={addBetObject} gameData={game} homeData={homeData} awayData={awayData} />

      if (gameState === "F" || gameState === "O" || gameState === "D") {
        finalGames.push(outputBlock)
      } else if (gameState === "I") {
        liveGames.push(outputBlock)
      } else {
        scheduledGames.push(outputBlock)
      }
      return
    })
  }

  let matchupBlocks = (liveGames.length !== 0 || scheduledGames.length !== 0 || finalGames.length !== 0) ? (
    <React.Fragment>
      {
        liveGames.length !== 0  && (
          <div className="gamesSection">
            <h3 className="gamesSection__header">Live</h3>
            {liveGames}
          </div>
        )
      }
      {
        scheduledGames.length !== 0 && (
          <div className="gamesSection">
            <h3 className="gamesSection__header">Upcoming</h3>
            {scheduledGames}
          </div>
        )
      }
      {
        finalGames.length !== 0 && (
          <div className="gamesSection">
            <h3 className="gamesSection__header">Completed</h3>
            {finalGames}
          </div>
        )
      }
    </React.Fragment>
  ) : (
    <div className="noGamesSection">
      <img src={leagueLogo} />
      <h2>No Games Scheduled</h2>
    </div>
  )
  // console.log(betObjects,'betObjects')
  // console.log(betObjectsTemp,'betObjectsTemp')
  return (
    <div className="App" style={{backgroundColor:'#f0f0f0',minHeight:'100vh',display:'flex',flexFlow:'column'}}>
      <header className="App-header">
        <div className="App-header__content">
          <div>
            <h1>baseball-bets</h1>
            <div className="project-tag">
              <span>Check out the project</span>
              <a href="http://github.com/cgilroy/baseball-bets">
                <img src={github} />
              </a>
            </div>
          </div>
          {betSummary}
        </div>
      </header>
      <div style={{display:'flex',flex:'1 1 auto',flexFlow:'column',padding: '15px',maxWidth:'900px', width:'100%', margin:'0 auto',justifyContent:'center'}}>
        {
          loading ? (
            <div style={{display:'flex',alignItems:'center',justifyContent:'center',height:'100%'}}>
              <div className="lds-ripple"><div></div><div></div></div>
            </div>
          ) : matchupBlocks
        }
      </div>
    </div>
  );
}

export default App;
