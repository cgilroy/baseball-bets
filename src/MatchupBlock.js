import React, {useState, useEffect} from 'react'
import './MatchupBlock.css'
import Card from '@material-ui/core/Card';
import Paper from '@material-ui/core/Paper';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import 'typeface-roboto';
import RunComparison from './RunComparison.js'
import SummarySection from './SummarySection.js'
import Checkmark from './resources/checkmark.svg'
import Xmark from './resources/x-mark.svg'
import ExpandMoreIcon from './resources/expand-more.svg'
import Moment from 'react-moment'

const FetchLiveData = (gamePk,callback) => {
  let gameUrl = 'https://statsapi.mlb.com/api/v1.1/game/'+gamePk+'/feed/live'
  fetch(gameUrl).then(resp => resp.json()).then(data => {
    // console.log(data,'FetchLiveData');
    let inningData = data.liveData.linescore.inningState + ' ' + data.liveData.linescore.currentInningOrdinal
    callback(inningData)
  })
}

const MatchupBlock = (props) => {
  const [dropDownActive, setDropDownActive] = useState(false)
  const [inningData, setInningData] = useState()
  const gameState = props.gameData.status.codedGameState
  // console.log(props,'props')
  let comparisonResult = RunComparison(props.homeData, props.awayData,props.gameData.dayNight)
  // console.log(comparisonResult,'comparisonResult')
  let homeBlock
  let awayBlock
  let homeScore = props.gameData.teams.home.score
  let awayScore = props.gameData.teams.away.score

  const doneFetch = (data) => {
    setInningData(data)
  }

  const gameUrl = props.gameData.gamePk
  if (gameState === 'I') FetchLiveData(gameUrl,doneFetch);

  useEffect(() => {
    if (comparisonResult.winner !== '') {
      // console.log('addBetObject')
      let winningGame = homeScore > awayScore ? 'HOME' : awayScore > homeScore ? 'AWAY' : ''
      let betState = winningGame === comparisonResult.winner ? 'WIN' : winningGame !== '' ? 'LOSE' : 'TIE'
      props.addBetObject({type: gameState, state: betState})
    }
  },[props.gameData])

  if (gameState === 'F' || gameState === 'O') {
    homeBlock = (
      <Card className={(comparisonResult.winner === 'HOME') ? ('card card__winner') : ('card')}>
        <CardContent className={(awayScore < homeScore) ? ('card__content card__content--game-winner') : ('card__content card__content--game-loser')} style={{paddingBottom:'16px'}}>
          {
            (comparisonResult.winner === 'HOME' && awayScore < homeScore) ? (
              <img className='checkmarkIcon' src={Checkmark} style={{marginRight:'8px'}}/>
            ) : (comparisonResult.winner === 'HOME' && awayScore > homeScore) ? (
              <img className='checkmarkIcon' src={Xmark} style={{marginRight:'8px'}}/>
            ) : ''
          }
          <img src={`https://www.mlbstatic.com/team-logos/${props.homeData.id}.svg`} className='team-logo' />
          {props.homeData.name}
        </CardContent>
      </Card>
    )
    awayBlock = (
      <Card className={(comparisonResult.winner === 'AWAY') ? ('card card__winner') : ('card')}>
        <CardContent className={(awayScore > homeScore) ? ('card__content card__content--game-winner') : ('card__content card__content--game-loser')} style={{paddingBottom:'16px'}}>
          {
            (comparisonResult.winner === 'AWAY' && awayScore > homeScore) ? (
              <img className='checkmarkIcon' src={Checkmark} style={{marginRight:'8px'}}/>
            ) : (comparisonResult.winner === 'AWAY' && awayScore < homeScore) ? (
              <img className='checkmarkIcon' src={Xmark} style={{marginRight:'8px'}}/>
            ) : ''
          }
          <img src={`https://www.mlbstatic.com/team-logos/${props.awayData.id}.svg`} className='team-logo' />
          {props.awayData.name}
        </CardContent>
      </Card>
    )
  } else {
    homeBlock = (
      <Card className={(comparisonResult.winner === 'HOME') ? ('card card__winner') : ('card')}>
        <CardContent className='card__content' style={{paddingBottom:'16px'}}>
          <img src={`https://www.mlbstatic.com/team-logos/${props.homeData.id}.svg`} className='team-logo' />
          {props.homeData.name}
        </CardContent>
      </Card>
    )
    awayBlock = (
      <Card className={(comparisonResult.winner === 'AWAY') ? ('card card__winner') : ('card')}>
        <CardContent className='card__content' style={{paddingBottom:'16px'}}>
          <img src={`https://www.mlbstatic.com/team-logos/${props.awayData.id}.svg`} className='team-logo' />
          {props.awayData.name}
        </CardContent>
      </Card>
    )
  }

  const toggleDropDown = () => {
    setDropDownActive(!dropDownActive)
  }
  return(
    <Paper style={{padding:'15px 15px 0 15px',margin:'10px 0',cursor:'default'}}>
      <TimeData gameState={gameState} inningData={inningData} gameDate={props.gameData.gameDate} location={props.gameData.venue.name} />
      <div className='matchup'>
        {homeBlock}
        <CenterScore gameState={gameState} gameData={props.gameData} comparisonResult={comparisonResult} />
        {awayBlock}
      </div>
      <div className='dropDown'>
        <CardActions style={{width:'100%',justifyContent:'center',paddingTop:'0'}}>
          <IconButton
            className={(dropDownActive) ? 'expanded' : 'collapsed'}
            onClick={toggleDropDown}
            aria-expanded={dropDownActive}
            aria-label="Show more"
          >
            <img src={ExpandMoreIcon} style={{height:'15px'}}/>
          </IconButton>
        </CardActions>
        <SummarySection data={comparisonResult.summary} active={dropDownActive}/>
      </div>
    </Paper>
  )
}

const TimeData = (props) => {
  let markup = ''
  switch (props.gameState) {
    case 'F':
    case 'O':
      markup = <span>FINAL</span>
      break;
    case 'D':
      markup = <span>DELAYED</span>
      break;
    case 'I':
      markup = (
        <React.Fragment>
          <div>
            <span style={{color:'#259b24'}}>{props.inningData}</span>
            <div className="liveIndicator">
              <div className="liveIndicator__bar"></div>
            </div>
          </div>
        </React.Fragment>
      );
      break;
    default:
      markup = (
        <React.Fragment>
          <span><Moment format="h:mm A">{props.gameDate}</Moment></span>
          <span style={{paddingLeft:'15px'}}>{props.location}</span>
        </React.Fragment>
      );
  }
  return (
    <div style={{display:'flex',alignItems:'center',marginBottom:'10px',textAlign:'left'}}>
      {markup}
    </div>
  )
}

const CenterScore = (props) => {

  const markup = (props.gameState === "I" || props.gameState === "F" || props.gameState === "O") ? (
    <React.Fragment>
      <div className='win-count'>
        <div>
          <span style={{fontWeight:'bolder',fontSize:'24px'}}>{props.gameData.teams.home.score}</span>
        </div>
      </div>
      <Typography style={{fontWeight:'bolder'}}>-</Typography>
      <div className='win-count'>
        <div>
          <span style={{fontWeight:'bolder',fontSize:'24px'}}>{props.gameData.teams.away.score}</span>
        </div>
      </div>
    </React.Fragment>
  ) : (
    <React.Fragment>
      <div className='win-count'>
        <div>
          <span>WINS</span>
        </div>
        <div>
          <span style={{fontWeight:'bolder'}}>{props.comparisonResult.score.home}</span>
        </div>
      </div>
      <Typography>VS</Typography>
      <div className='win-count'>
        <div>
          <span>WINS</span>
        </div>
        <div>
          <span style={{fontWeight:'bolder'}}>{props.comparisonResult.score.away}</span>
        </div>
      </div>
    </React.Fragment>
  )

  return (
    <div className='centerScore'>
      {markup}
    </div>
  )
}

export default MatchupBlock
