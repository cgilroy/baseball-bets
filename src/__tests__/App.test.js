import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';
import MatchupBlock from '../MatchupBlock'
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import matchupData from '../sample-data/matchup-block-data.json'

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('App', () => {
  it('App renders without crashing', () => {
    shallow(<App />)
  })
})

it('renders correctly', () => {
  const AppComponent = renderer.create(
    shallow(
      <MatchupBlock
        key={matchupData.gameData.gamePk}
        gameData={matchupData.gameData}
        homeData={matchupData.homeData}
        awayData={matchupData.awayData}
        />
    )
  ).toJSON();
  expect(AppComponent).toMatchSnapshot();
})
