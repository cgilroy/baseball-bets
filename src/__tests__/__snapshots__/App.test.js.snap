// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
  style={
    Object {
      "backgroundColor": "#f0f0f0",
      "display": "flex",
      "flexFlow": "column",
      "minHeight": "100vh",
    }
  }
>
  <header
    className="App-header"
  >
    <h1>
      baseball-bets
    </h1>
  </header>
  <div
    style={
      Object {
        "display": "flex",
        "flex": "1 1 auto",
        "flexFlow": "column",
        "justifyContent": "center",
        "margin": "0 auto",
        "maxWidth": "900px",
        "padding": "15px",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "height": "100%",
          "justifyContent": "center",
        }
      }
    >
      <div
        className="lds-ripple"
      >
        <div />
        <div />
      </div>
    </div>
  </div>
</div>
`;
